### PR TITLE
Wildfly client model integration and quickstart (client)

### PR DESCRIPTION
**Pull request by mskacelik**
/cc @jmartisk

### CODE REVIEW

File: quickstart-client/README.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# GraphQL Client quickstart 
+
+## Prerequisites
+- JDK 1.8+
```
Comment (jmartisk) [inline quickstart-client/README.md:4]: The current requirement is 11+ afaik

File: client-subsystem/src/main/java/org/wildfly/extension/microprofile/graphql/client/ClientModelsExtension.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.Any;
+import jakarta.enterprise.inject.Default;
+import jakarta.enterprise.inject.spi.AfterBeanDiscovery;
+import jakarta.enterprise.inject.spi.BeanManager;
+import jakarta.enterprise.inject.spi.Extension;
+import jakarta.enterprise.util.AnnotationLiteral;
+import jakarta.inject.Singleton;
+
+public class ClientModelsExtension implements Extension {
```
Comment (jmartisk) [inline client-subsystem/src/main/java/org/wildfly/extension/microprofile/graphql/client/ClientModelsExtension.java:13]: Please add a javadoc comment what this is for (that it registers `ClientModels` as a singleton CDI bean)

File: quickstart-client/src/main/java/org/wildfly/extras/quickstart/microprofile/graphql/client/FilmResource.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        final Integer port = Integer.getInteger("port", 8080);
+        url = "http://localhost:" + port + "/quickstart/graphql";
+    }
+
+    @GET
+    @Path("/typesafe/films")
+    @Produces(MediaType.APPLICATION_JSON)
+    public List<Film> getAllFilms() {
+        System.out.println("Using URL " + url);
+        FilmClientApi client = TypesafeGraphQLClientBuilder.newBuilder()
```
Comment (jmartisk) [inline quickstart-client/src/main/java/org/wildfly/extras/quickstart/microprofile/graphql/client/FilmResource.java:45]: I think we could inject `ClientModels` in this class and pass it to the builders

File: client-subsystem/src/main/java/org/wildfly/extension/microprofile/graphql/client/deployment/MicroProfileGraphQLClientDependencyProcessor.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import static org.wildfly.extension.microprofile.graphql.client.deployment.MicroProfileGraphQLClientDeploymentProcessor.GRAPHQL_CLIENT_API;
+
+public class MicroProfileGraphQLClientDependencyProcessor implements DeploymentUnitProcessor {
+
+    @Override
+    public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        final DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+        final CompositeIndex compositeIndex = deploymentUnit.getAttachment(Attachments.COMPOSITE_ANNOTATION_INDEX);
+        if(!compositeIndex.getAnnotations(GRAPHQL_CLIENT_API).isEmpty()) {
+            MicroProfileGraphQLClientLogger.LOGGER.multipleGraphQLApiAnnotations();
```
Comment (jmartisk) [inline client-subsystem/src/main/java/org/wildfly/extension/microprofile/graphql/client/deployment/MicroProfileGraphQLClientDependencyProcessor.java:40]: I don't get this, this says that we found "multiple" annotations, but will appear even if there's just one, no? What's the point of this logging message?

Comment (mskacelik) [reply client-subsystem/src/main/java/org/wildfly/extension/microprofile/graphql/client/deployment/MicroProfileGraphQLClientDependencyProcessor.java:40]: Ok, I had on my mind "multiple" as in one or more annotations.

File: pom.xml [hunk 5]
```diff
@@ -51,7 +51,7 @@
              the built feature pack. -->
         <version.io.smallrye.graphql>2.9.1</version.io.smallrye.graphql>
         <version.org.eclipse.microprofile.graphql>2.0</version.org.eclipse.microprofile.graphql>
-        <version.com.graphql-java>22.1</version.com.graphql-java>
+        <version.com.graphql-java>22.2</version.com.graphql-java>
```
Comment (jmartisk) [inline pom.xml:54]: Let's remove this for now please, we'll do this along with upgrading to sr-gql 2.10.0 next week. Otherwise, all looks good to me!

**Review by jmartisk — APPROVED** [review]

**Pull request merged**